src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UpToDate/UpToDateCheckStatePersistence.cs

```diff
@@ -143,16 +153,54 @@ public Task StoreStateAsync(string projectPath, IImmutableDictionary<string, str
                 cancellationToken);
         }
 
+        private async Task EnsureDataInitializedAsync(CancellationToken cancellationToken)
+        {
+            if (_cacheFilePath is null || _dataByConfiguredProject is null)
+            {
+                string filePath = await GetCacheFilePathAsync(cancellationToken);
+
+                // Switch to a background thread before doing file I/O
+                await TaskScheduler.Default;
+
+                if (_cacheFilePath is null || _dataByConfiguredProject is null)
+                {
+                    _cacheFilePath = filePath;
+                    _dataByConfiguredProject = Deserialize(_cacheFilePath);
+                }
+            }
+
+            return;
```

Comment:
This shouldn't be needed, correct?

Reply:
Correct. It's a convention used in several places to separate the method's statements from any local functions. The idea is it makes it easier to see that there are no statements hidden later on in the method, as it would be otherwise valid to have statements after local functions.